### PR TITLE
Volunteer Map: Prefer more recent sign-ups

### PR DESCRIPTION
**Pull request by islemaster**
[PLC-554](https://codedotorg.atlassian.net/browse/PLC-554): Our [local volunteer search](https://code.org/volunteer/local) has a usability problem:

- We have volunteer sign-ups from 4+ years ago.
- Our search returns a limited number of results.
- Results are ordered oldest-first.

In combination, this means that on crowded areas of the map, newer sign-ups are never showing up in search results.

We've discussed a number of approaches to this problem, and decided the lowest lift for now is to reverse the sort order of local results so that the newest sign-ups show up first.

## Testing story

I've added a unit test that catches this behavior, and the one existing test did not break.  I refactored them both and they now clear out volunteer submissions in the test database before each test, so I believe they're better isolated and more accurately test what they say they do.

Manual testing is a bit tricky on this feature. Here are the steps I took:

1. Get secrets
   In particular, you need `google_maps_client_id` and `google_maps_secret` which can be defined in your locals.yml or configured to be pulled from the secrets manager in `development.yml.erb`.  I did the former.

2. Sign up volunteers
   Use the form at http://localhost.code.org:3000/volunteer to submit several possible volunteers in the same region, with names making their sign-up ordering obvious.

3. Process forms
   Run `bin/cron/process_forms` locally which uses the Google Maps API to geocode the submissions from step 2 and associate the metadata required for them to show up on the local volunteer map.

4. Visit the map
   Visit the map at http://localhost.code.org:3000/volunteer/local and check the region where you added volunteers.  Verify the expected ordering of volunteers in the list on the left.

### Before

image

### After

image

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: pegasus/test/test_form_routes.rb [hunk 1]
```diff
@@ -30,30 +30,58 @@ class FormRoutesTest < SequelTestCase
...
+      # Delete all volunteer sign-ups in the test database before each test
+      ::PEGASUS_DB[:forms].where(kind: 'VolunteerEngineerSubmission2015').delete
+
+      Pegasus.stubs(:logger)
+      stubs(:dashboard_user)
+      stubs(:request).returns(stub(ip: '1.2.3.4'))
+    end
+
+    it 'returns local results' do
+      create_volunteer name: 'Local Person', location: '37.774929,-122.419416'
```
Comment (hacodeorg) [inline pegasus/test/test_form_routes.rb:44]: These coordinates could be an Easter egg :)

**Review by breville — APPROVED** [review]
This is great!

File: pegasus/test/test_form_routes.rb [hunk 2]
```diff
@@ -30,30 +30,58 @@ class FormRoutesTest < SequelTestCase
...
+      results = search location: '37.774368,-122.428760'
+      assert_equal 0.8236209090344097, results.first['distance']
```
Comment (hacodeorg) [inline pegasus/test/test_form_routes.rb:46]: Could you add an explanation for this number? A precise comparison to a float number like this seems really strict.

**Review by hacodeorg — APPROVED** [review]
LGTM. Appreciate the test instruction, it'd be helpful to other people working on this code later.

File: pegasus/test/test_form_routes.rb (hunk 1 above)
Comment (islemaster) [reply pegasus/test/test_form_routes.rb:44]: Market and Van Ness in San Francisco? I'm not sure what they're an easter egg for.

File: pegasus/test/test_form_routes.rb (hunk 2 above)
Comment (islemaster) [reply pegasus/test/test_form_routes.rb:46]: I'm not sure actually - it's unchanged from the existing test that was here.  I think it's just a check that our location-based search gives consistent results.

Comment (wjordan) [reply pegasus/test/test_form_routes.rb:46]: I added this test in 31a362001ddc3d81c9056bc9995dbc1fe2d11b02 (part of #22645) - it's just arbitrary (but valid) coordinates to ensure consistency of the API's distance calculation. I thought it was self-explanatory enough as 'fixture' data, but I wouldn't object if someone wanted to add a comment explanation to clarify further (or change the coordinates to something more easter-egg like).

**Pull request merged**